### PR TITLE
[TypeDeclaration] Handle infered ThisType on auto import on ReturnTypeDeclarationRector

### LINKED ISSUE #6978: Incorrect behavior of ReturnTypeDeclarationRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | 75c0b27d37ee34844b0eb5faf35742c6631d5ab2 (0.12.15)      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/1ec82e17-2d17-6468-a0cb-6d13244aef63

```php
<?php

namespace T1 {
    class A {
        /**
         * @return $this
         */
        public function itself()
        {
        }
    }
}

namespace T2 {
    use T1\A;
    
    class B {
        public function test(): A 
        {
            return (new A)->itself();
        }
    }
}
```

### Responsible rules

* `ReturnTypeDeclarationRector`

## Expected Behavior

Rector should not do anything to `B::test()`. Only happens with `AUTO_IMPORT_NAMES` set to true.

### PR DESCRIPTION
**Pull request by samsonasik**
Closes #1753 
Fixes https://github.com/rectorphp/rector/issues/6978

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: .github/workflows/rector.yaml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
             -
                 # commit only to core contributors who have repository access
                 if: github.event.pull_request.head.repo.full_name == github.repository
-                uses: EndBug/add-and-commit@v8.0.1
+                uses: EndBug/add-and-commit@v7.5.0
```
Comment (TomasVotruba) [inline .github/workflows/rector.yaml:71]: Is this on purpose?

Comment (TomasVotruba) [reply .github/workflows/rector.yaml:71]: I see https://github.com/rectorphp/rector-src/pull/1756#issuecomment-1027597913 :)

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**